File: 프로그래머스/pauly00/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    count = 0
    zeroCount = 0
    
    while s != "1":
        zeroCount +=s.count('0')
        
        s = s.replace('0', '')
        s = bin(len(s))[2:] # 2진수 변환
        
        count +=1
        
    return [count, zeroCount]

File: 프로그래머스/HeeYeon-Ko/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        
        String result = s;
        int zero = 0;
        int count = 0;
        
        while(!result.equals("1")){
            int beforeLength = result.length();
            result = result.replace("0", "");
            int afterLength = result.length();
            zero += beforeLength - afterLength;
            
            result = Integer.toString(afterLength, 2);
            count++;
        }
        
        answer[0] = count;
        answer[1] = zero;
        
        return answer;
    }
}